pkg/internal/codegen/parse/index.go

```diff
@@ -183,61 +115,47 @@ type resourceTags struct {
-	result := resourceTags{}
+//parseResourceAnnotationValue is a helper function to extract resource annotation.
+func parseResourceAnnotationValue(tag string, count int) (resourceTags, int, error) {
+	res := resourceTags{}
 	for _, elem := range strings.Split(tag, ",") {
-		kv := strings.Split(elem, "=")
-		if len(kv) != 2 {
-			log.Fatalf("// +kubebuilder:resource: tags must be key value pairs.  Expected "+
-				"keys [path=<subresourcepath>] "+
+		key, value, err := general.ParseKV(elem)
+		if err != nil {
+			return resourceTags{}, count, fmt.Errorf("// +kubebuilder:resource: tags must be key value pairs.  Expected "+
+				"keys [path=<resourcepath>] "+
 				"Got string: [%s]", tag)
 		}
-		value := kv[1]
-		switch kv[0] {
-		//case "rest":
-		//	result.REST = value
+		switch key {
 		case "path":
-			result.Resource = value
-		//case "strategy":
-		//	result.Strategy = value
+			res.Resource = value
 		case "shortName":
-			result.ShortName = value
+			res.ShortName = value
+		default:
+			return resourceTags{}, count, fmt.Errorf("The given input %s is invalid", value)
 		}
 	}
-	return result
+	count++
+	return res, count, nil
 }
 
-// ParseSubresourceTag parses the tags in a "+subresource=" comment into a subresourceTags struct
-//func parseSubresourceTag(c *codegen.APIResource, tag string) subresourceTags {
-//	result := subresourceTags{}
-//	for _, elem := range strings.Split(tag, ",") {
-//		kv := strings.Split(elem, "=")
-//		if len(kv) != 2 {
-//			log.Fatalf("// +subresource: tags must be key value pairs.  Expected "+
-//				"keys [request=<requestType>,rest=<restImplType>,path=<subresourcepath>] "+
-//				"Got string: [%s]", tag)
-//		}
-//		value := kv[1]
-//		switch kv[0] {
-//		case "request":
-//			result.RequestKind = value
-//		case "rest":
-//			result.REST = value
-//		case "path":
-//			// Strip the parent resource
-//			result.Path = strings.Replace(value, c.Resource+"/", "", -1)
-//		}
-//	}
-//	return result
-//}
-
-// getResourceTag returns the value of the "+resource=" comment tag
-func (b *APIs) getResourceTag(c *types.Type) string {
-	comments := Comments(c.CommentLines)
-	resource := comments.getTag("resource", ":") + comments.getTag("kubebuilder:resource", ":")
-	if len(resource) == 0 {
-		panic(errors.Errorf("Must specify +kubebuilder:resource comment for type %v", c.Name))
+// parseResourceAnnotation parses the tags in a "+resource=" comment into a resourceTags struct.
+func parseResourceAnnotation(t *types.Type) (resourceTags, error) {
+	finalResult := resourceTags{}
+	var count, res int
+	for _, comment := range t.CommentLines {
+		anno := general.GetAnnotation(comment, "kubebuilder:resource")
+		if len(anno) == 0 {
+			continue
+		}
+		result, temp, err := parseResourceAnnotationValue(anno, count)
+		if err != nil {
+			return resourceTags{}, err
+		}
+		res++
+		if res != temp {
+			return resourceTags{}, fmt.Errorf("resource annotation should only exists once per type")
+		}
```

Comment:
As stated in previous comment,  you just need to track if `resourceAnnotationFound` which can be toggled once `parseResourceAnnotationValue` returns success i.e. no-error. and if we encounter another resource annotation and we were able to parse it suceesfully and `resourceAnnotationFound` was already true, then we can throw an error that multiple resource annotations are not allowed.

Reply:
Follow my previous comments. I've included your feedback in this code.